authors/apps/articles/views.py

```diff
@@ -238,9 +244,82 @@ def delete(self, request, **kwargs):
             status.HTTP_200_OK
         )
 
+class ShareOnFaceBookView(APIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (JSONRenderer, )
+
+
+    def post(self, request, *args, **kwargs):
+        server_url = f"https://{get_current_site(request)}"
+        articles_endpoint = 'api/articles'
+        facebook_share_link = f"{server_url}/{articles_endpoint}/{kwargs.get('article_id')}/"
+        facebook_share_url = "https://www.facebook.com/sharer/sharer.php?u="
+        message = {"facebook_share_link": f"{facebook_share_url}{facebook_share_link}"}
+        return Response(message)
+
+class ShareOnTwitterView(APIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (JSONRenderer, )
+
+    def post(self, request, *args, **kwargs):
+        server_url = f"https://{get_current_site(request)}"
+        articles_endpoint = 'api/articles'
+        twitter_share_link = f"{server_url}/{articles_endpoint}/{kwargs.get('article_id')}/"
+        twitter_share_url = "https://twitter.com/home?status="
+        message = {"twitter_share_link": f"{twitter_share_url}{twitter_share_link}"}
+        return Response(message)
```

Comment:
I notice some duplication in the two post methods under ShareOnFacebookView and ShareOnTwitterView. Could those two methods be merged into one?